# Analysez des données de systèmes éducatifs

## Sommaire

- [Partie 1](#Partie-1)
  - Étape 1 : Chargez les données  
  - Étape 2 : Collectez des informations basiques sur chaque jeu de données  
  - Étape 3 : Réalisez votre premier nettoyage  

- [Partie 2](#Partie-2)
  - Étape 1 : Réduisez le périmètre en utilisant une approche métier  
  - Étape 2 : Réduisez le périmètre en utilisant une approche data  
  - Étape 3 : Consolidez vos résultats dans un dataframe (pays, indicateurs)  

- [Partie 3](#Partie-3)
  - Étape 1 : Identifiez les indicateurs redondants d’un point de vue statistique  
  - Étape 2 : Analysez vos indicateurs restants et formulez une liste de pays pertinents pour l’implantation d’une academy  
  - Étape 3 : Formalisez vos résultats  
  - Étape 4 : Rendez vos analyses et vos résultats reproductibles par vos collègues  


## Partie 1 

### Étape 1 : Chargez les données  


Nous avons cinq données que nous allons charger pour voir ce qu’elles contiennent.  
Ces données sont :

- **EdStatsCountry.csv**
- **EdStatsCountry-Series.csv**
- **EdStatsData.csv**
- **EdStatsFootNote.csv**
- **EdStatsSeries.csv**


#### Donnée 1 EdStatsCountry.csv


In [1]:
# chargement du fichier
import pandas as pd
data = pd.read_csv("EdStatsCountry.csv")


In [2]:
# afficher les 5 premières lignes
display(data.head())

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

            IMF data dissemination standard  \
0                                       NaN   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3  General Data Dissemination System (GDDS)   
4                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   
2                    NaN               NaN                         2005   
3                 2010.0            2012.0                         2006   
4                    NaN            2006.0                          NaN   

  Unnamed: 31  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 32 columns]

Se qu'on peux remarquer dans se tableau sont:
Ce tableau fournit des informations générales et statistiques sur chaque pays.

- `Country Code` : le code ISO à trois lettres qui identifie chaque pays
- `Short Name` : le nom court utilisé pour désigner le pays
- `Long Name / Table Name`: le nom officiel complet du pays
- `2-alpha code` : le code ISO à deux lettres
- `Currency Unit` : la monnaie utilisée dans le pays
- `Region` : la zone géographique à laquelle le pays appartient
- `Income Group` : le niveau de revenu du pays (ex. : *Low income*, *High income*)
- `IMF data dissemination standard` : indique la qualité du partage des données économiques selon le FMI
- `Latest population census` : année du dernier recensement de population
- `Latest trade data` : année des dernières données commerciales disponibles
- `Unnamed: 31` : probablement une colonne vide ou non renseignée


Remarque : ce tableau est utile pour **filtrer les pays selon leur niveau de données disponibles ou leur profil économique**.


#### Donnée 2 EdStatsCountry-Series.csv


In [3]:
# chargement du fichier
import pandas as pd
data = pd.read_csv("EdStatsCountry-Series.csv")


In [4]:
# afficher les 5 premières lignes
display(data.head())

CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

Se qu'on peux voir dans se tableau sont:

- `CountryCode` : code ISO du pays 
- `SeriesCode` : identifiant de l’indicateur 
- `DESCRIPTION` : source ou méthode de calcul 
- `Unnamed: 3` : colonne vide 

Remarque : ce tableau permet de **comprendre quels indicateurs sont disponibles pour chaque pays**.



#### Donnée 3 EdStatsData.csv


In [5]:
# chargement du fichier
import pandas as pd
data = pd.read_csv("EdStatsData.csv")


In [6]:
# afficher les 5 premières lignes
display(data.head())

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2060  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

Se qu'on peut voir dans se tableau sont: 
Chaque ligne représente un indicateur d'inscription scolaire pour une région (ici : Arab World), sur une série d'années.
- `Country Name` : nom du pays ou de la région 
- `Country Code` : code ISO 
- `Indicator Name` : nom de l’indicateur 
- `Indicator Code` : identifiant de l’indicateur 
- `Colonnes de dates` (1970 à 2090) 
Remarque : la majorité des valeurs sont manquantes (`NaN`), ce qui limite l’analyse directe.


#### Donnée 4 EdStatsFootNote.csv


In [7]:
# chargement du fichier
import pandas as pd
data = pd.read_csv("EdStatsFootNote.csv")


In [8]:
# afficher les 5 premières lignes
display(data.head())

CountryCode      SeriesCode    Year          DESCRIPTION  Unnamed: 4
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.         NaN
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.         NaN
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.         NaN
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.         NaN
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.         NaN

Se qu'on peut voir dans se tableau c'est: 
- `CountryCode` : code ISO du pays
- `SeriesCode` : identifiant de l’indicateur
- `Year` : année de la donnée
- `DESCRIPTION` : commentaire sur la donnée (ici : "Country estimation.")
- `Unnamed: 4` : colonne vide 

#### Donnée 5 EdStatsSeries.csv


In [9]:
# chargement du fichier
import pandas as pd
data = pd.read_csv("EdStatsSeries.csv")


In [10]:
# afficher les 5 premières lignes
display(data.head())

Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   
2  BAR.NOED.15UP.FE.ZS  Attainment   
3     BAR.NOED.15UP.ZS  Attainment   
4  BAR.NOED.2024.FE.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   
2  Barro-Lee: Percentage of female population age...   
3  Barro-Lee: Percentage of population age 15+ wi...   
4  Barro-Lee: Percentage of female population age...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   
2  Percentage of female population age 15+ with n...   
3  Percentage of population age 15+ with no educa...   
4  Percentage of female population age 20-24 with...   

                                     Long definition  Unit of measure  \
0  Percentage of female population age 15-19 with...              NaN   
1  Percentage of population age 15-19 with no edu...              NaN   
2  Percentage of female population age 15+ with n...              NaN   
3  Percentage of population age 15+ with no educa...              NaN   
4  Percentage of female population age 20-24 with...              NaN   

  Periodicity Base Period Other notes Aggregation method  ...  \
0         NaN         NaN         NaN                NaN  ...   
1         NaN         NaN         NaN                NaN  ...   
2         NaN         NaN         NaN                NaN  ...   
3         NaN         NaN         NaN                NaN  ...   
4         NaN         NaN         NaN                NaN  ...   

  Notes from original source  General comments  \
0                        NaN               NaN   
1                        NaN               NaN   
2                        NaN               NaN   
3                        NaN               NaN   
4                        NaN               NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   
2  Robert J. Barro and Jong-Wha Lee: http://www.b...   
3  Robert J. Barro and Jong-Wha Lee: http://www.b...   
4  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   
2                                 NaN                   NaN   
3                                 NaN                   NaN   
4                                 NaN                   NaN   

  Related source links Other web links  Related indicators  License Type  \
0                  NaN             NaN                 NaN           NaN   
1                  NaN             NaN                 NaN           NaN   
2                  NaN             NaN                 NaN           NaN   
3                  NaN             NaN                 NaN           NaN   
4                  NaN             NaN                 NaN           NaN   

   Unnamed: 20  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  

[5 rows x 21 columns]

Se qu'on peut voir dans se tableau sont:
Ce tableau décrit chaque indicateur statistique lié à l’éducation, avec ses métadonnées et définitions.

- `Series Code` : identifiant unique de l’indicateur 
- `Topic` : thème général
- `Indicator Name` : nom complet de l’indicateur
- `Short definition` / `Long definition` : définitions synthétique et détaillée
- `Unit of measure`, `Periodicity`, `Base Period` : souvent vides ici
- `Source` : origine des données 
- Autres colonnes : commentaires, liens, pertinence, licence, etc.

Remarque : ce tableau sert de **référentiel pour interpréter les codes d’indicateurs** présents dans les autres jeux de données.


### Étape 2 : Collectez des informations basiques sur chaque jeu de données  


Dans cette étape, nous allons explorer et nettoyer chaque fichier de données.  
Nous allons analyser la structure des datasets, détecter les doublons et valeurs manquantes, et produire un jeu de données propre.

**voici se qu'on va faire** 

1. Définir ce que représente **une ligne** du fichier  
2. Calculer le **nombre de lignes et de colonnes**  
3. Identifier et compter les **doublons**  
4. Supprimer les **doublons** détectés  
5. Calculer la **proportion de valeurs manquantes** par colonne  
6. Supprimer les **colonnes inutilisables**  
7. Pour les colonnes **numériques** 
8. Pour les colonnes **catégorielles**







#### Donnée 1 EdStatsCountry.csv

##### 1.Définir ce que représente une ligne du fichier

In [4]:
# chargement du fichier
import pandas as pd
data = pd.read_csv("EdStatsCountry.csv")


In [5]:
# afficher les 5 premières lignes
display(data.head())

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

            IMF data dissemination standard  \
0                                       NaN   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3  General Data Dissemination System (GDDS)   
4                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   
2                    NaN               NaN                         2005   
3                 2010.0            2012.0                         2006   
4                    NaN            2006.0                          NaN   

  Unnamed: 31  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 32 columns]

In [6]:
df.info()

NameError: name 'df' is not defined

<span style="color:red; font-weight:bold;">⚠️ Attention :</span> Selon les informations recherchées sur internet, cette erreur se produit à l'import du jeu de données. Il faut remplacer `data` par `df`.


In [16]:
import pandas as pd
df = pd.read_csv("EdStatsCountry.csv")

In [17]:
display(df.head())

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

            IMF data dissemination standard  \
0                                       NaN   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3  General Data Dissemination System (GDDS)   
4                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   
2                    NaN               NaN                         2005   
3                 2010.0            2012.0                         2006   
4                    NaN            2006.0                          NaN   

  Unnamed: 31  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 32 columns]

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 32 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                       214 non-null    object 
 9   WB-2 code 

Se que on peut observer 
- Le dataset contient **241 lignes** et **32 colonnes**, chaque ligne correspondant probablement à un pays.
- La majorité des colonnes sont de type **object** (texte/catégoriel), seules 4 colonnes sont numériques (`float64`).
- Certaines colonnes présentent **beaucoup de valeurs manquantes** :
  - `Unnamed: 31` est entièrement vide.
  - `National accounts reference year` et `Latest industrial data` ont beaucoup de valeurs manquantes.
- Quelques colonnes sont presque complètes : `Country Code`, `Short Name`, `Table Name`, `Long Name`.
- D’autres colonnes très peu renseignées (`Other groups`, `Alternative conversion factor`) pourraient être **inutilisables pour l’analyse**.

**Conclusion :** Le dataset est suffisamment complet pour certaines informations de base sur les pays, mais plusieurs colonnes devront être nettoyées ou supprimées avant toute analyse.


##### 2.Calculer le nombre de ligne et de colonne

In [3]:
print("Dimensions :", df.shape)

Dimensions : (241, 32)


##### 3.Identifier et compter les doublons

In [4]:
print("Doublons :", df.duplicated().sum())

Doublons : 0


##### 4 .Supprimer les doublons

Dans la jeu de donnée EdStatsCountry.csv il n'y a pas de doublon donc aucun a supprimer 

##### 5 .Calculer la proportion de valeurs manquantes par colonne

In [7]:
print("Valeurs manquantes (%) :")
print(df.isnull().mean() * 100)

Valeurs manquantes (%) :
Country Code                                           0.000000
Short Name                                             0.000000
Table Name                                             0.000000
Long Name                                              0.000000
2-alpha code                                           1.244813
Currency Unit                                         10.788382
Special Notes                                         39.834025
Region                                                11.203320
Income Group                                          11.203320
WB-2 code                                              0.414938
National accounts base year                           14.937759
National accounts reference year                      86.721992
SNA price valuation                                   18.257261
Lending category                                      40.248963
Other groups                                          75.933610
System of Natio

#### Donnée 2 EdStatsCountry-Series.csv

##### 1.Définir ce que représente une ligne du fichier

In [9]:
import pandas as pd
df = pd.read_csv("EdStatsCountry-Series.csv")

In [10]:
display(df.head())

CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB


Se qu'on l'on observe
- **Nombre de lignes :** 613  
- **Nombre de colonnes :** 4  

**Colonnes :**  
- `CountryCode` : complète, type texte (`object`)  
- `SeriesCode` : complète, type texte (`object`)  
- `DESCRIPTION` : complète, type texte (`object`)  
- `Unnamed: 3` : entièrement vide (0 valeurs), type numérique (`float64`)  



##### 2.Calculer le nombre de ligne et de colonne

In [12]:
print("Dimensions :", df.shape)

Dimensions : (613, 4)


##### 3.Identifier et compter les doublons

In [13]:
print("Doublons :", df.duplicated().sum())

Doublons : 0


##### 4 .Supprimer les doublons

Dans la jeu de donnée EdStatsCountry-Series.csv il n'y a pas de doublon donc aucun a supprimer 

##### 5 .Calculer la proportion de valeurs manquantes par colonne

In [14]:
print("Valeurs manquantes (%) :")
print(df.isnull().mean() * 100)

Valeurs manquantes (%) :
CountryCode      0.0
SeriesCode       0.0
DESCRIPTION      0.0
Unnamed: 3     100.0
dtype: float64


#### Donnée 3 EdStatsData.csv

##### 1.Définir ce que représente une ligne du fichier

In [19]:
import pandas as pd
df = pd.read_csv("EdStatsData.csv")

In [20]:
display(df.head())

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2060  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983      

##### 2.Calculer le nombre de ligne et de colonne

In [22]:
print("Dimensions :", df.shape)

Dimensions : (886930, 70)


##### 3.Identifier et compter les doublons

In [26]:
# Affiche un booléen pour chaque ligne : True si la ligne est un doublon
df.duplicated()


0         False
1         False
2         False
3         False
4         False
          ...  
886925    False
886926    False
886927    False
886928    False
886929    False
Length: 886930, dtype: bool

In [23]:
print("Doublons :", df.duplicated().sum())

Doublons : 0


##### 4 .Supprimer les doublons

##### 5 .Calculer la proportion de valeurs manquantes par colonne

In [24]:
print("Valeurs manquantes (%) :")
print(df.isnull().mean() * 100)

Valeurs manquantes (%) :
Country Name        0.000000
Country Code        0.000000
Indicator Name      0.000000
Indicator Code      0.000000
1970               91.849639
                     ...    
2085               94.200670
2090               94.200670
2095               94.200670
2100               94.200670
Unnamed: 69       100.000000
Length: 70, dtype: float64


#### Donnée 4 EdStatsData.csv

##### 1.Définir ce que représente une ligne du fichier

##### 2.Calculer le nombre de ligne et de colonne

##### 3.Identifier et compter les doublons

##### 4 .Supprimer les doublons

##### 5 .Calculer la proportion de valeurs manquantes par colonne

#### Donnée 5 EdStatsData.csv

##### 1.Définir ce que représente une ligne du fichier

##### 2.Calculer le nombre de ligne et de colonne

##### 3.Identifier et compter les doublons

##### 4 .Supprimer les doublons

##### 5 .Calculer la proportion de valeurs manquantes par colonne

### Étape 3 : Réalisez votre premier nettoyage

Dans cette étape, nous allons explorer et nettoyer les fichiers liés aux pays et aux séries de données.  
Nous allons identifier les faux pays, filtrer les lignes incorrectes et produire des dataframes cohérents entre eux.

**Voici ce qu'on va faire**
1. Examiner le fichier **Country** pour détecter les **faux pays**.
2. Compter le **nombre de lignes et de colonnes** dans chaque dataframe.
3. Identifier et supprimer les **lignes correspondant aux faux pays** dans `Country`.
4. Pour les autres dataframes (`Country-Series`, `FootNote`, `Data`) :
   - Stocker les faux pays dans une liste et filtrer avec `isin()` **ou**
   - Utiliser un **inner join** avec le dataframe `Country` nettoyé.
5. Vérifier que les dataframes filtrés restent **cohérents** entre eux.
6. Sauvegarder les versions **nettoyées** des dataframes pour l’analyse suivante.


#### Donnée 1 EdStatsCountry.csv

#### Donnée 2 EdStatsCountry-Series.csv

#### Donnée 3 EdStatsData.csv

#### Donnée 4 EdStatsData.csv

#### Donnée 5 EdStatsData.csv

## Partie 2

### Étape 1 : Réduisez le périmètre en utilisant une approche métier

### Étape 2 : Réduisez le périmètre en utilisant une approche data

### Étape 3 : Consolidez vos résultats dans un dataframe (pays, indicateurs)

## Partie 3

### Étape 1 : Identifiez les indicateurs redondants d’un point de vue statistique

### Étape 2 : Analysez vos indicateurs restants et formulez une liste de pays pertinents pour l’implantation d’une academy

### Étape 3 : Formalisez vos résultats

### Étape 4 : Rendez vos analyses et vos résultats reproductibles par vos collègues